### PR TITLE
some plugin developer tips and tricks

### PR DESCRIPTION
**Pull request by raymondnijssen**

### CODE REVIEW

File: source/docs/pyqgis_developer_cookbook/plugins.rst [hunk 1]
```diff
@@ -573,3 +573,21 @@ You should see your plugin in the correct language.
...
+
+Tips and Tricks
+===============
+
+Plugin Reloader
+---------------
+
+During development of your plugin you will frequently need to reload it in QGIS for testing. This is
+very easy using the Plugin Reloader plugin. You can find it as an experimental plugin with the Plugin
+Manager.
```
Comment (ghtmtt) [inline source/docs/pyqgis_developer_cookbook/plugins.rst]: what about adding a link to the plugin.qgis.org website?

File: source/docs/pyqgis_developer_cookbook/plugins.rst [hunk 2]
```diff
@@ -573,3 +573,21 @@ You should see your plugin in the correct language.
...
+
+During development of your plugin you will frequently need to reload it in QGIS for testing. This is
+very easy using the Plugin Reloader plugin. You can find it as an experimental plugin with the Plugin
+Manager. 
+
+Accessing Plugins
+-----------------
+
+You can access all installed plugins from within QGIS using python, which can be handy for debugging
+purposes.::
```
Comment (ghtmtt) [inline source/docs/pyqgis_developer_cookbook/plugins.rst]: I'll add something like: `You can access all the classes of the installed plugins`.. Anyway, that's a good new entry to have in the cookbook!

Comment (Gustry) [reply source/docs/pyqgis_developer_cookbook/plugins.rst]: +1, I would add what we can do with this snippet.

Thanks for sharing this!

**Comment by raymondnijssen** [comment]
What about pulling this for now (it is not wrong) and adding extra info later? I did the edits online on github and don't know how to re-edit them

**Comment by DelazJ** [comment]
@raymondnijssen 
> I did the edits online on github and don't know how to re-edit them

Clicking the pencil at the right of the filename at https://github.com/qgis/QGIS-Documentation/pull/2774/files should do the trick or https://github.com/raymondnijssen/QGIS-Documentation/blob/patch-1/source/docs/pyqgis_developer_cookbook/plugins.rst
Hope that helps

File: source/docs/pyqgis_developer_cookbook/plugins.rst [hunk 3]
```diff
@@ -573,3 +573,30 @@ You should see your plugin in the correct language.
...
+
+You can access all the classes of installed plugins from within QGIS using python,
+which can be handy for debugging purposes.::
+
+	my_plugin = qgis.utils.plugins['My Plugin']
+
+Share your plugin
+-----------------
+QGIS is hosting hundreds of plugins in the plugin repository. Consider sharing
+yours! It will extent the possibilities of QGIS and people will be able to learn
```
Comment (DelazJ) [inline source/docs/pyqgis_developer_cookbook/plugins.rst]: extend

Comment (raymondnijssen) [reply source/docs/pyqgis_developer_cookbook/plugins.rst]: done

**Comment by rduivenvoorde** [comment]
@DelazJ can this be merged now?

**Pull request merged**